cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpCredentialConnector.java

```diff
@@ -72,16 +66,148 @@ public CloudCredentialStatus create(AuthenticatedContext authenticatedContext) {
+     *                    AuthenticatedContext.
+     *                    If the passed Optional is empty, then a default
+     *                    message is going to be passed to the status
+     *                    instance.
+     * @return The combined CloudCredentialStatus instance which stores all
+     * the necessary/required data for a proper object with a FAILED status.
+     */
+    private CloudCredentialStatus createFailedCloudCredentialStatusWithExc(Exception e, AuthenticatedContext authContext,
+            Optional<String> message) {
+        String logMessage = String.format("Could not verify credential, detailed message: %s", e.getMessage());
+        LOGGER.warn(logMessage, e);
+        return new CloudCredentialStatus(
+                authContext.getCloudCredential(),
+                CredentialStatus.FAILED,
+                e,
+                message.orElse("Could not verify credential!"));
+    }
+
+    /**
+     * Checks the validity of the provided GcpContext's credential. It sends
+     * a http request through the google's http api which validates the data
+     * set provided from the given GcpContext.
+     *
+     * @param gcpContext the GcpContext instance which credential would be
+     *                   checked.
+     * @throws IOException if something happens while listing the regions,
+     *                     this exception would thrown by the api.
+     */
     private void preCheckOfGooglePermission(GcpContext gcpContext) throws IOException {
+        /* The actual response which comes from the method invocation is ignored, because if
+        the credential is invalid or the request has timed out or something other happens
+        an exception would thrown automatically */
+        gcpContext.getCompute().regions().list(gcpContext.getProjectId()).executeUsingHead();
+    }
+
+    /**
+     * Attempts to get the "error_description" parameter's value from the given
+     * TokenResponseException's content. If there is no "error_description"
+     * parameter, or it has no value or something occurs during the process, an
+     * empty Optional would return.
+     *
+     * @param e The TokenResponseException which content should have a
+     *          "error_description" parameter with a string value.
+     * @return A String Optional with the content of the "error_description"
+     * from the exception, or an empty one.
+     */
+    private Optional<String> getErrDescriptionFromTokenResponse(TokenResponseException e) {
         try {
-            gcpContext.getCompute().regions().list(gcpContext.getProjectId());
-        } catch (NullPointerException ignore) {
-            LOGGER.error(String.format("Google authentication failed for project-id '%s' because of a null value:", gcpContext.getProjectId()), ignore);
+            return getErrDescriptionFromTokenErrorJson(new JsonParser().parse(e.getContent()).getAsJsonObject());
+        } catch (Exception err) {
```

Comment:
The parse never throws checked Exception, so not need to catch it. It throws JsonParseException and IllegalStateException (or simply RuntimeException).